libvast/src/logger.cpp

```diff
@@ -15,9 +15,255 @@
 
 #include "vast/config.hpp"
 #include "vast/defaults.hpp"
+#include "vast/error.hpp"
+#include "vast/si_literals.hpp"
 #include "vast/system/configuration.hpp"
+#include "vast/detail/assert.hpp"
 
 #include <caf/atom.hpp>
+#include <caf/local_actor.hpp>
+
+#include <cassert>
+#include <memory>
+
+#include <spdlog/async.h>
+#include <spdlog/common.h>
+#include <spdlog/sinks/ansicolor_sink.h>
+#include <spdlog/sinks/null_sink.h>
+#include <spdlog/sinks/rotating_file_sink.h>
+
+namespace vast {
+
+caf::optional<caf::detail::scope_guard<void (*)()>>
+create_log_context(const vast::invocation& cmd_invocation,
+                   const caf::settings& cfg_file) {
+  if (!vast::detail::setup_spdlog(cmd_invocation, cfg_file))
+    return {};
+
+  return {caf::detail::make_scope_guard(
+    std::addressof(vast::detail::shutdown_spdlog))};
+}
+
+namespace {
+
+constexpr bool is_vast_loglevel(const int value) {
+  switch (value) {
+    case VAST_LOG_LEVEL_QUIET:
+    case VAST_LOG_LEVEL_ERROR:
+    case VAST_LOG_LEVEL_WARNING:
+    case VAST_LOG_LEVEL_INFO:
+    case VAST_LOG_LEVEL_VERBOSE:
+    case VAST_LOG_LEVEL_DEBUG:
+    case VAST_LOG_LEVEL_TRACE:
+      return true;
+  }
+  return false;
+}
+
+/// Converts a vast log level to spdlog level
+constexpr spdlog::level::level_enum vast_loglevel_to_spd(const int value) {
+  VAST_ASSERT(is_vast_loglevel(value));
```

Comment:
since using `VAST_ASSERT` I have this compiler warning
```
/Users/a4z/work/cpp/vast/libvast/src/logger.cpp:65:3: warning: uninitialized variable in a constexpr function is a C++20 extension [-Wc++2a-extensions]
  VAST_ASSERT(!is_vast_loglevel(value));
  ^
/Users/a4z/work/cpp/vast/libvast/vast/detail/assert.hpp:35:15: note: expanded from macro 'VAST_ASSERT'
        void* vast_array[10];                                                  \
              ^
```

Reply:
I removed the constexpr , once this moves to C++20, `constexpr bool is_constant_evaluated` can be used to have either `VAST_ASSERT` or a `static_assert`